### PR TITLE
tests: Add deployment with LB scale test

### PR DESCRIPTION
**Pull request by ionutbalutoiu**
#### What type of PR is this?

/sig apps
/sig windows

#### What this PR does / why we need it:

Add a test case with a deployment behind a simple load balancer whose address is being constantly hit via HTTP requests.

The test passes if there are no errors when doing HTTP requests to the load balancer address, during deployment scale up / down operations.

#### Which issue(s) this PR fixes:

N/A

#### Special notes for your reviewer:

N/A

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/e2e/apps/deployment.go [hunk 1]
```diff
@@ -168,6 +173,12 @@ var _ = SIGDescribe("Deployment", func() {
 		e2eskipper.SkipUnlessAtLeast(len(nodes.Items), 3, "load-balancer test requires at least 3 schedulable nodes")
 		testRollingUpdateDeploymentWithLocalTrafficLoadBalancer(f)
 	})
+	ginkgo.It("should not disrupt a load-balancer's connectivity during scaling", func() {
```
Comment (andrewsykim) [inline test/e2e/apps/deployment.go:176]: This test might be a better fit in https://github.com/kubernetes/kubernetes/blob/master/test/e2e/network/loadbalancer.go

File: test/e2e/apps/deployment.go [hunk 2]
```diff
@@ -1442,6 +1453,112 @@ func testRollingUpdateDeploymentWithLocalTrafficLoadBalancer(f *framework.Framew
...
+func testScaleDeploymentWithLoadBalancer(f *framework.Framework) {
+	ns := f.Namespace.Name
+	c := f.ClientSet
+
+	name := "test-scale-with-lb"
+	podLabels := map[string]string{"name": name}
+	initialReplicas := int32(2)
+
+	framework.Logf("Creating Deployment %q", name)
+	d := e2edeployment.NewDeployment(name, initialReplicas, podLabels, AgnhostImageName, AgnhostImage, appsv1.RollingUpdateDeploymentStrategyType)
```
Comment (andrewsykim) [inline test/e2e/apps/deployment.go:1465]: Using a DaemonSet here might provide better coverage -- specifically the case where a node goes from 1 -> 0 endpoints. If you use a Deployment, the rolling update might happen in a single node where the node health check never fails.

**Review by andrewsykim — COMMENTED** [review]
/ok-to-test

**Comment by ionutbalutoiu** [comment]
Hello @andrewsykim,

Thank-you for the review comments. All make sense to me.

I opened PR https://github.com/kubernetes/kubernetes/pull/114052 which adds the same test to `test/e2e/network/loadbalancer.go` using a DaemonSet resource.

**Pull request closed without merging**